foundation/foundation-mda/src/main/resources/templates/data-delivery-pyspark/encryption.py.vm

```
@@ -28,14 +28,17 @@
             # If that property is missing then we should ignore the policy.
             if encrypt_policy.encryptPhase:
                 if self.step_phase.lower() == encrypt_policy.encryptPhase.lower():
-
+                #if (${step.hasInboundNativeCollectionTypeAndRelations()})
```

q: could we move this up and fail faster?

---

There is a trade off with moving the exception up.  In it's current position it will only fail if the policy applies to a specific pipeline step, but I suppose it wouldn't hurt if it fails for all pipeline steps(?)

---

Now that I think about it - if we move it up the user won't have to wait for pre-processing to complete before finding out that the encryption will cause an exception.  Good catch